### ISSUE TITLE
Angular Deferrable Views not working when used in custom library component

### ISSUE THREAD

**Issue by praveshShetty** [angular/angular#53936]
### Which @angular/* package(s) are the source of the bug?

core

### Is this a regression?

No

### Description

I'm trying to implement deferrable views in a custom library component and use it in my main project, but what I'm seeing is:

1. The deferred component is being fetched in the main library bundle itself and is not being split into another bundle
2. The placeholder element appears for a split second and then the view disappears. On checking the HTML, it seems like the deferred component is not rendered correctly. I cannot see the child elements of the deferred component it dev tools. I have attached a reference image. The components hd-base-article-email, hd-base-top-categories, hd-base-related-articles are all deferred components.

<img width="487" alt="Screenshot 2024-01-10 at 6 55 46 PM" src="https://github.com/angular/angular/assets/128784467/f8bf643f-2da5-46db-b302-ee668e4b4df5"> 

### Please provide the exception or error you saw

```true
no visible errors in the console
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Library Repo:

Angular CLI: 17.0.9
Node: 18.18.0
Package Manager: pnpm 8.10.3
OS: darwin arm64

Angular: 17.0.8
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.7
@angular-devkit/build-angular   17.0.9
@angular-devkit/core            17.0.9
@angular-devkit/schematics      17.0.9 (cli-only)
@angular/cdk                    17.0.4
@angular/cli                    17.0.9
@angular/material               17.0.4
@schematics/angular             17.0.9 (cli-only)
ng-packagr                      17.0.3
rxjs                            6.6.0
typescript                      5.2.2
zone.js                         0.14.2

Main Project Repo:

Angular CLI: 17.0.2
Node: 18.18.0
Package Manager: pnpm 8.10.3
OS: darwin arm64

Angular: 17.0.2
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, language-service, material, platform-browser
... platform-browser-dynamic, platform-server, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1700.2 (cli-only)
@angular-devkit/build-angular   17.0.2
@angular-devkit/core            17.0.2 (cli-only)
@angular-devkit/schematics      17.0.2 (cli-only)
@angular/google-maps            17.0.4
@angular/localize               17.0.8
@angular/pwa                    0.1102.8
@angular/service-worker         16.2.8
@nguniversal/builders           16.2.0
@nguniversal/express-engine     16.2.0
@schematics/angular             17.0.2 (cli-only)
rxjs                            6.6.7
typescript                      5.2.2
zone.js                         0.13.3
```

### Anything else?

I have already tried to implement deferrable views in components inside the main repo itself and it seems to be working fine. Issue is only when used in library repo.

Here are the things which I have followed as a requirement:

1. Using angular 17 in both the main project and library project
2. Using on viewport condition to defer the block
3. The component inside the defer block is a standalone component is not used anywhere outside the defer block and also not referenced using viewchild

**Comment by JoostK** [comment]
This is currently expected; libraries are compiled differently to applications and do not currently account for defer loading of views.

This should be included in the docs.

> The placeholder element appears for a split second and then the view disappears. On checking the HTML, it seems like the deferred component is not rendered correctly. I cannot see the child elements of the deferred component it dev tools. I have attached a reference image. The components hd-base-article-email, hd-base-top-categories, hd-base-related-articles are all deferred components.

Could you please share a runnable reproduction that showcases this issue.

**Comment by praveshShetty** [comment]
Sure. I have created a minimum reproduction of the issue in this project [https://github.com/praveshShetty/deferrable-views-reproduction/tree/main](https://github.com/praveshShetty/deferrable-views-reproduction/tree/main) along with the steps to reproduce. Wasn't completely sure on how to reproduce this on stackblitz as it requires building of a library and using it , so have created it on github itself.  Hope this helps !!

**Comment by ahmed-a-ibrahim** [comment]
I am using NX workspace and the defer in NX lib, then used it in the demo app, and it works fine.

**Comment by JoostK** [comment]
> I am using NX workspace and the defer in NX lib, then used it in the demo app, and it works fine.

This is expected for Nx's non-buildable libraries, as those are compiled as part of the application build and not independently as library.

**Comment by mauriziocescon** [comment]
Ok, good to know! Got the same issue on a big enterprise app where everything is coming from already built ng-libs. 

I wonder: any technical issue to have the case covered in the near future? Or unlikely to happen?

**Comment by AndrewKushnir** [comment]
Support for `@defer` in libraries has landed in PR #54908 and will be available as a part of v18 release.
